File: examples/pub_connect.lua
#!/usr/bin/env luajit
-- Connect PUB socket to tcp://localhost:5813
-- Publish timestamped uuid's on heartbeat topic
require "sys"
require "zhelpers"
-- require lzmq module as local variable zmq
local zmq = require "lzmq"
-- socket.gettime() has higher precision than os.time()
local socket = require "socket"
local uuid = require "uuid"
local json = require "cjson"
-- see also example at uuid.seed()
uuid.randomseed(socket.gettime()*10000)
-- get ZeroMQ context
local context = zmq.context()
-- set ZeroMQ PUB socket
local publisher, err = context:socket{zmq.PUB, connect = "tcp://localhost:5813"}
zassert(publisher, err)
-- struct message
local message = {}
-- connect process loop
while true do
  -- time in seconds, relative to the origin of the universe. 
  message['timestamp'] = socket.gettime()
  -- set message uuid
  message['uuid'] = uuid()
  -- Send message to the subscriber
  message = "heartbeat " .. json.encode(message)
  -- socket send message
  publisher:send(message)
  -- process sleep
  sys.sleep(0.500)
end

File: examples/pub_connect.py
#!/usr/bin/env python

import zmq
from zmq.eventloop import ioloop
from zmq.eventloop.future import Context, Poller
from tornado import gen
import ujson as json
import arrow
import uuid

ioloop.install()

@gen.coroutine
def publisher(port=5813):
    context = Context()
    pub_uuid = str(uuid.uuid4())
    pub = context.socket(zmq.PUB)
    pub.connect("tcp://zmq.nonsense.ws:%s" % port)
    poller = Poller()
    poller.register(pub, zmq.POLLOUT)
    while True:
        topic = 'heartbeat'
        utc = arrow.utcnow()
        raw = json.dumps({"timestamp":str(utc.timestamp), "uuid": pub_uuid})
        message = '{0} {1}'.format(topic, raw)
        yield pub.send(message)
        yield gen.sleep(1)

def main():
    loop = ioloop.IOLoop.instance()
    loop.add_callback(publisher)
    loop.start()

if __name__ == '__main__':
    main()
